hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/WritableTopicStore.java

```diff
@@ -59,11 +63,57 @@ public WritableTopicStore(@NonNull final WritableStates states) {
     public void put(@NonNull final Topic topic) {
         requireNonNull(topicState);
         requireNonNull(topic);
-        topicState.put(topic.topicNumber(), asMerkleTopic(topic));
+        topicState.put(EntityNum.fromLong(topic.topicNumber()), asMerkleTopic(topic));
     }
 
+    /**
+     * Commits the changes to the underlying data storage.
+     * TODO: Not sure if the stores have responsibility of committing the changes. This might change in the future.
```

Comment:
I think this is the right place for this. What other class(es) would commit topic changes if not `WritableTopicStore`? That seems like the only purpose for this class ???

Reply:
I am not sure.